### PR TITLE
Screenshot fix

### PR DESCRIPTION
**Pull request by tapir**
First 2 commits implements PNG save method and switches r_screenshot to it.
Other commits remove r_screenshot_quality and jpg mentions as they are not needed anymore.

This commit removes dependency to jpeglib and since jpg is patent-encumbered I think it's a good thing to remove it. We don't lose any functionality. No body uses jpg textures and for screenshots there is not much disk space penalty.

I did not remove jpeglib from build tools project files (Xcode, autotools etc...) as I have not much experiance with them.

### CODE REVIEW

**Comment by tapir** [comment]
This also fixes #166

File: src/game/default/g_item.c [hunk 1]
```diff
@@ -882,7 +882,7 @@ void G_PrecacheItem(const g_item_t *it) {
 		else if (g_str_has_suffix(data, ".wav") || g_str_has_suffix(data, ".ogg"))
 			gi.SoundIndex(data);
 		else if (g_str_has_suffix(data, ".tga") || g_str_has_suffix(data, ".png")
-				|| g_str_has_suffix(data, ".jpg") || g_str_has_suffix(data, ".pcx"))
```
Comment (jdolan) [inline src/game/default/g_item.c]: Assets might still reference .jpg textures, and SDL_image will still load .jpg just fine for us.

File: src/image.c [hunk 2]
```diff
@@ -61,7 +33,7 @@ static _Bool img_palette_initialized;
 #define AMASK 0xff000000
 
 // image formats, tried in this order
-static const char *img_formats[] = { "tga", "png", "jpg", "wal", "pcx", NULL };
```
Comment (jdolan) [inline src/image.c]: Hm, again, I think i'd like to retain the ability to load .jpg, just in case. It's free.

File: src/image.c [hunk 3]
```diff
@@ -206,49 +178,29 @@ void Img_ColorFromPalette(uint8_t c, vec_t *res) {
...
 
-	jpeg_destroy_compress(&cinfo);
-
-	fclose(f);
+	SDL_FreeSurface(ss);
+	Mem_Free(buffer);
+	SDL_FreeRW(f);
 	return true;
-}
+}
```
Comment (jdolan) [inline src/image.c]: 💯

File: src/tools/quemap/qzip.c [hunk 4]
```diff
@@ -94,7 +94,7 @@ static void AddSound(const char *sound) {
  * a warning will be issued should we fail to resolve the specified image.
  */
 static void AddImage(const char *image, _Bool required) {
-	const char *image_formats[] = { "tga", "png", "jpg", "pcx", "wal", NULL };
+	const char *image_formats[] = { "tga", "png", "pcx", "wal", NULL };
```
Comment (jdolan) [inline src/tools/quemap/qzip.c]: Again, please retain the ability to load .jpg.

**Comment by tapir** [comment]
Done :+1:

**Pull request merged**